### DISCUSSION TITLE
Show ufo fold markings in statuscol

### DISCUSSION THREAD

**Discussion by jborbik** [LazyVim/LazyVim#1881] [Q&A] [answered]
Currently in statuscol the extmarks are being filtered to show only marks and git signs on the left side. Is there a way to pass there nvim-ufo fold marks as well?

**Comment by radvil** [comment]
U can refer to mine
```lua
  {
    "kevinhwang91/nvim-ufo",
    dependencies = {
      "kevinhwang91/promise-async",
      {
        "luukvbaal/statuscol.nvim",
        opts = function()
          local builtin = require("statuscol.builtin")
          return {
            relculright = true,
            ft_ignore = {
              "DiffviewFiles",
              "dashboard",
              "NvimTree",
              "neo-tree",
              "Outline",
              "Trouble",
            },
            segments = {
              { text = { builtin.foldfunc, " " }, click = "v:lua.ScFa" },
              { text = { builtin.lnumfunc, " " }, click = "v:lua.ScLa" },
              { text = { "%s" }, click = "v:lua.ScSa" },
            },
          }
        end,
      },
    },
    keys = {
      {
        "zR",
        function()
          require("ufo").openAllFolds()
        end,
        desc = "UFO » Open all folds",
      },
      {
        "zr",
        function()
          require("ufo").openFoldsExceptKinds()
        end,
        desc = "UFO » Open folds excepts kinds",
      },
      {
        "zM",
        function()
          require("ufo").closeAllFolds()
        end,
        desc = "UFO » Close all folds",
      },
      {
        "[z",
        function()
          require("ufo").goPreviousClosedFold()
          vim.schedule(function()
            require("ufo").peekFoldedLinesUnderCursor()
          end)
        end,
        desc = "UFO » Peek prev fold",
      },
      {
        "]z",
        function()
          require("ufo").goNextClosedFold()
          vim.schedule(function()
            require("ufo").peekFoldedLinesUnderCursor()
          end)
        end,
        desc = "UFO » Peek next fold",
      },
    },

    opts = {
      provider_selector = function(_, filetype, _)
        return ({
          typescript = { "treesitter", "indent" },
          html = { "treesitter", "indent" },
          python = { "indent" },
          vim = "indent",
          git = "",
          Outline = "",
        })[filetype]
      end,
    },
  },
```

**Reply by jborbik** [reply]
Thank you for your effort! But I'd rather stick to the statuscolumn from LazyVim which works perfectly fine, only nvim-ufo is missing for me :)

**Reply by radvil** [reply]
Well I thought u re not using the latest release like me.

**Comment by jborbik** [comment, marked as answer]
It was actually easier than I thought :D I had to change https://github.com/LazyVim/LazyVim/blob/7831fc94ca5989baf766c0bb6ad36a70838c3d5a/lua/lazyvim/util/ui.lua#L93 to `local components = { "%C", "", "", "" } -- foldcolumn, left, middle, right`and indices below this line accordingly.

**Reply by farzadmf** [reply]
@Jendker if you don't mind me asking: how are you changing that? Are you using a fork of `LazyVim`?

**Reply by jborbik** [reply]
No, I just copied the part which I needed, modified it the way I liked it and included it in my config: https://github.com/Jendker/dotfiles/blob/9db3959bc7b5d01d31d737597e295646de45dfd8/nvim/lua/settings.lua#L160-L329

**Reply by farzadmf** [reply]
Ah right, basically copy/pasted the whole `statuscolumn` logic it seems.

Since it seems like you're using `nvim-ufo`, I was wondering if you have this issue with it:
I use folds a lot, and doing `zO` expands current fold recursively (only the current one).

Unfortunately, `nvim-ufo` doesn't seem to have anything similar, and the issue I have is that it keeps closing my folds (since, if I'm not mistaken, it only keeps the folds expanded with its own functions). Do you have/did you have a similar issue?

**Reply by jborbik** [reply]
No, I stopped using nvim-ufo because of various errors I couldn’t solve so unfortunately I cannot help. I’m not familiar enough with nvim-ufo.

**Reply by farzadmf** [reply]
Makes sense; not sure why it is so popular; I gave it a try for 10 minutes, and I gave up on it!
